tests/test_examples.py

```diff
@@ -0,0 +1,37 @@
+import os.path
+from openapi_spec_validator import validate_v2_spec_url
+from textwrap import dedent
+
+import pytest
+import yaml
+
+# TODO make it find automatically with importlib and import only inside fixture
+from examples.extended_example_with_marshmallow import spec
+
+# TODO
+def locate_example_outputs(example_dir):
+    pass
+
+@pytest.fixture()
+def spec_and_expected_output(request):
+    spec = request.param.spec
+    output_file = request.param.output_path
+    output_file_path = os.path.join("examples", output_file)
+    abs_output_path = os.path.abspath(output_file_path)
+    return spec.to_yaml(), open(abs_output_path).read()
+
+SpecOutputPath = namedtuple('SpecOutputPath', ['spec', 'output_path'])
+
+@pytest.mark.parameterize(['spec_and_expected_output'], [
+    pytest.param(SpecOutputPath(spec, 'extended_example_with_marshmallow_output.yaml'), id='extended_example')
+], indirect=['spec_and_expected_output'])
+def test_extended_example_with_marshmallow(spec_and_expected_output):
+    spec, expected = spec_and_expected_output
+    
+    assert yaml.load(spec) == yaml.load(expected)
+
+@pytest.mark.parameterize(['spec_and_expected_output'], [
+    pytest.param('examples/extended_example_with_marshmallow_output.yaml', id='extended_example')
+], indirect=['spec_and_expected_output'])
+def test_example_outputs_valid_openapiv2(uri):
+    validate_v2_spec_url("file://" + uri)
```

Comment:
This seems sensible and was basically what I was envisaging. The main difference in my mind would be inlining the contents of `spec_and_expected_output` in the test function, as I'm not sure it needs to be a fixture (I think it could also just be a function that gets called at the start of the test?), but I'm happy to defer to your preferences on this.

Reply:
I'm fine with both. Just split both tests as I did and don't make the elements used for the test hardcoded (allows, IMO, better organization and naming).
If you rather loading the files in the test function, that's fine for me too.